api-get-conditions.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * Minio Go Library for Amazon S3 Compatible Cloud Storage (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package minio
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// GetConditions - get conditions implement methods for setting
+// conditions for a GetObject request.
+// http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectGET.html
+type GetConditions struct {
+	http.Header
+}
```

Comment:
But PreConditions are generic header information ... i.e it can apply to HEAD too if required.

`SetRange` does not sound right for a "Conditions" type.

You could name it just `Header`